blocks/library/reusable-block/index.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Server-side rendering of the `core/reusable-block` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the `core/reusable-block` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Rendered HTML of the referenced block.
+ */
+function gutenberg_render_block_core_reusable_block( $attributes ) {
+	$reusable_blocks = get_posts( array(
+		'post_type' => 'gb_reusable_block',
+		'name' => $attributes['ref'],
+	) );
+
+	if ( empty( $reusable_blocks ) ) {
+		return '';
+	}
+
+	$reusable_block = array_shift( $reusable_blocks );
+
+	$blocks = gutenberg_parse_blocks( $reusable_block->post_content );
+
+	if ( empty( $blocks ) ) {
+		return '';
+	}
+
+	$block = array_shift( $blocks );
+	$block_name = $block['blockName'];
+	$block_attributes = $block['attrs'];
+	$block_content = $block['rawContent'];
+
+	$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+	if ( $block_type ) {
+		return $block_type->render( $block_attributes, $block_content );
+	} else {
+		return $block_content;
+	}
```

Comment:
I wonder if we can factorize some code between here and the `do_blocks` function into a `render_block( block )` function

Reply:
Great catch! Cleaned up in ccf3024e1f317f918463e956f967e949e0514b51 🌈